iamy/cfn.go

```diff
@@ -13,7 +13,7 @@ import (
 	"github.com/aws/aws-sdk-go/service/cloudformation/cloudformationiface"
 )
 
-var cfnResourceRegexp = regexp.MustCompile(`-[A-Z0-9]{10,20}$`)
+var cfnResourceRegexp = regexp.MustCompile(`^(?=.{10,20})(?=.*[A-Z])(?=.*[0-9A-Z]).*$`)
```

Comment:
I think it'd be better to have a separate regex for elasticbeanstalk managed buckets, and have a separate check in https://github.com/envato/iamy/blob/main/iamy/cfn.go#L138-L140

It looks like they're in the format of `elasticbeanstalk-$region-$accountid`. Also wondering if the matcher for this should be tightened up to match the initial `elasticbeanstalk-` as well as the `-$accountid`... https://docs.aws.amazon.com/elasticbeanstalk/latest/dg/AWSHowTo.S3.html

Reply:
Easy enough, not entirely sure it's worth trying to support every weird s3 bucket naming scheme we come across.

Things ending in account numbers are pretty common (we've used the scheme in several places).
I'm not trying to change this to detect beanstalk buckets, I'm trying to make sure we don't capture account numbers when we go looking for a cfn id suffix...

Reply:
So even if we specifically exclude the beanstalk bucket in this example we'll still need to fix the regular expression as it catches things that are not cfn suffix ids.   Beanstalk is just the one we happened to notice..

Reply:
I imagine a separate regex for `endsInAccountId` could be created, then a check earlier than the CFN regex would `return true` early. That should mean the cfn regex wouldn't need updating. Seems like an option. The main goal of that is to have simpler regex's so they're easier to understand.

Reply:
So far I've found
beanstalk, code-star, config, athena, sagemaker all using the same scheme 
as well as the envato-logs-<accountnumber> stuff we devised